gist/client.py

```diff
@@ -175,11 +177,22 @@ def terminal_width():
 
     """
     try:
-        exitcode = fcntl.ioctl(
-                0,
-                termios.TIOCGWINSZ,
-                struct.pack('HHHH', 0, 0, 0, 0))
-        h, w, hp, wp = struct.unpack('HHHH', exitcode)
+        if platform.system() == "Windows":
+            from ctypes import windll, create_string_buffer
+            hStdErr = -12
+            herr = windll.kernel32.GetStdHandle(hStdErr)
+            csbi = create_string_buffer(22)
```

Comment:
What does the 22 refer to?

Reply:
22 is the exact buffer size for receiving the output of `GetConsoleScreenBufferInfo()`

```python
>>> import struct
>>> struct.calcsize("hhhhHhhhhhh")
22
```

Reply:
Perhaps add that code instead of the magic number.